packages/core/src/types.ts

```diff
@@ -788,7 +790,8 @@ type MachineOptionsActions<
         TContext,
         Cast<Prop<TIndexedEvents, TEventsCausingActions[K]>, EventObject>,
         Cast<Prop<TIndexedActions, K>, BaseActionObject>
-      >;
+      >
+    | Array<Cast<Prop<TIndexedActions, K>, BaseActionObject>['type']>;
```

Comment:
We probably should accept more things here. For instance those should be valid:
```
group: [assign(), 'another']
group: [assign(), sendParent()]
group: [assign(), { type: 'another' }]
```

Reply:
I'm working on shifting it to:

```ts
{
  [K in keyof TEventsCausingActions]?:
    | ActionObject<
        TContext,
        Cast<Prop<TIndexedEvents, TEventsCausingActions[K]>, EventObject>
      >
    | ActionFunction<
        TContext,
        Cast<Prop<TIndexedEvents, TEventsCausingActions[K]>, EventObject>,
        Cast<Prop<TIndexedActions, K>, BaseActionObject>
      >
    | Array<
        | ActionObject<
            TContext,
            Cast<Prop<TIndexedEvents, TEventsCausingActions[K]>, EventObject>
          >
        | ActionFunction<
            TContext,
            Cast<Prop<TIndexedEvents, TEventsCausingActions[K]>, EventObject>,
            Cast<Prop<TIndexedActions, K>, BaseActionObject>
          >
        | Cast<Prop<TIndexedActions, K>, BaseActionObject>['type']
      >;
}
```

It'll accept objects and functions though it still leaves us with the same type problems as `pure`. At least they're consistent 😆

Reply:
Also thinking I should change `Cast<Prop<TIndexedActions, K>, BaseActionObject>['type']` to `keyof TEventsCausingActions`

- autocompletion of `type` strings
- the typegen portion of this automatically adds `type` strings in groups to `eventsCausingActions`, so it wouldn't do anything for you on the first use of a `type`, but if you used it in later groups you'd get autocomplete

Think there'd be any downsides to that?